### PR TITLE
WIP: Try to lint paket

### PR DESCRIPTION
**Pull request by forki**
Mhm doesn't work on mono!?
/cc @duckmatt

### CODE REVIEW

**Comment by duckmatt** [comment]
I had the same problem linting FSharpLint itself, the solution was to add the FSharpLint directory as an assembly search path: `#I @"packages/FSharpLint/"`

**Comment by forki** [comment]
thanks. Now I get https://travis-ci.org/fsprojects/Paket/builds/49967776#L1924

The file exists but it's a relative path and not found.

**Comment by duckmatt** [comment]
It seems like the paths in MSBuild are getting set relative to where FSharpLint is being run from (build.fsx) so `<SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\..\</SolutionDir>` is setting up `$(SolutionDir)` to `/home/travis/build/fsprojects/Paket/../../`

A quick fix would be to add a condition to the import so that it's not included if it can't be found:

    <Import Project="$(SolutionDir)\.paket\paket.targets" Condition="Exists('$(SolutionDir)\.paket\paket.targets')" />

I'll add an issue to FSharpLint to have a look into getting the MSBuild paths to be relative to the `.fsproj` files

**Comment by duckmatt** [comment]
It should hopefully now work without that quick fix in 0.1.15

File: build.fsx [hunk 1]
```diff
@@ -119,6 +122,11 @@ Target "Build" (fun _ ->
     |> ignore
 )
 
+Target "Lint" (fun _ ->
+    !! "**/*.fsproj" 
+    |> Seq.iter (FSharpLint (fun options -> { options with FailBuildIfAnyWarrnings = true }))
```
Comment (forki) [inline build.fsx]: @duckmatt I just noticed a typo. Parameter should be called `FailBuildIfAnyWarnings`

**Comment by forki** [comment]
and we have a new strange error: https://travis-ci.org/fsprojects/Paket/builds/50064698#L1897

**Comment by duckmatt** [comment]
I see there's also problems on the windows build, I'll clone the repo and get it running correctly for 0.1.16. Thanks for letting me know about these issues, I think this is the first time anyone has tried to integrate it into a large project other than it linting itself, so this will be a good test

**Comment by forki** [comment]
:shipit:

**Comment by duckmatt** [comment]
0.1.16 runs successfully on Paket now: https://github.com/duckmatt/Paket/tree/Lint 

Might be worth turning off `MaxCharactersOnLine` as that rule is generating a lot of errors - it's one I've been considering turning off by default. Let me know if there's anything I can do to help and if there's any changes you'd like to see to the linter

**Comment by forki** [comment]
ok I'll close this one. If you are interested please send your own PR with suggested fixes for Paket.

**Pull request closed without merging**